client/services/code_preprocessors/PythonCodePreprocessorServiceSpec.js

```diff
@@ -460,23 +462,30 @@ describe('PythonCodePreprocessorService', function() {
   describe('_generateBuggyOutputTestCode', function() {
     it('should add correct buggy output test code to skeleton code',
       function() {
-        var buggyOutputTests = [BuggyOutputTestObjectFactory.create({
-          buggyFunctionName: 'buggyFunc',
-          messages: ['a', 'b', 'c']
-        }), BuggyOutputTestObjectFactory.create({
-          buggyFunctionName: 'buggyFunc2',
-          messages: ['d', 'e', 'f']
-        })];
+        var buggyOutputTests = [
+            [BuggyOutputTestObjectFactory.create({
+              buggyFunctionName: 'buggyFunc',
+              messages: ['a', 'b', 'c']
+            }), BuggyOutputTestObjectFactory.create({
+              buggyFunctionName: 'buggyFunc2',
+              messages: ['d', 'e', 'f']
+            })]
+        ];
         var expectedGeneratedCode = [
           'def matches_buggy_function(func):',
           '    buggy_results = []',
-          '    for test_input in test_inputs:',
-          '        buggy_results.append(System.runTest(func, test_input))',
+          '    for tests in test_inputs:',
+          '        one_task_results = []',
+          '        for test in tests:',
+          '            one_task_results.append(System.runTest(func, test))',
+          '        buggy_results.append(one_task_results)',
           '    return buggy_results == correctness_test_results',
           '',
           'buggy_output_test_results = []',
-          'buggy_output_test_results.append(matches_buggy_function(buggyFunc))',
-          'buggy_output_test_results.append(matches_buggy_function(buggyFunc2))',
+          'one_task_buggy_output_test_results = []',
```

Comment:
The array stores the test results of **one single task only**, therefore in our opinion it's important to emphasize 'one task' in variable names.